intern/resource/boss/resource.py

```diff
@@ -442,6 +459,15 @@ def __init__(self, name, collection_name, experiment_name, type="image",
         self.exp_name = experiment_name
 
         self._type = self.validate_type(type)
+
+        # Class is considered fully initialized if datatype set during
+        # initialization.
+        if datatype == '':
```

Comment:
Seems like this is a little misleading. if `datatype` is set, that doesn't mean it necessarily is "fully initialized". Probably at a minimum should make sure everything that is needed for serialization is there?